### PR TITLE
fix(sto-bro): fix conditional value provided to FoldersMessageProvider

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes

Remove conditionally `undefined` value from `props` provided to `FoldersMessageProvider`

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-core/src/utils/createContextUtilities.tsx [hunk 1]
```diff
@@ -99,7 +99,11 @@ export default function createContextUtilities<
     throw new Error(INVALID_OPTIONS_MESSAGE);
   }
 
+  const contextDisplayName = `${contextName}Context`;
+  const providerDisplayName = `${contextName}Provider`;
+
   const Context = React.createContext<ContextType | undefined>(defaultValue);
+  Context.displayName = contextDisplayName;
```
Comment (calebpollman) [inline packages/react-core/src/utils/createContextUtilities.tsx:106]: Provides a display name for the context in dev tools

File: packages/react-storage/src/components/StorageBrowser/displayText/libraries/en/copyView.ts [hunk 2]
```diff
@@ -48,9 +48,6 @@ export const DEFAULT_COPY_VIEW_DISPLAY_TEXT: DefaultCopyViewDisplayText = {
       type: 'error',
     };
   },
-  getFolderSelectedMessage: (key: string) => {
-    return `Current folder selected: ${key}. There are no additional folders under this path.`;
-  },
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/displayText/libraries/en/copyView.ts:53]: Unused

File: packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CopyView/CopyViewProvider.tsx [hunk 3]
```diff
@@ -89,15 +87,6 @@ export function CopyViewProvider({
...
     : getActionCompleteMessage({ counts: statusCounts });
 
-  const foldersMessage = !hasFoldersInitialized
-    ? undefined
-    : getListFoldersResultsMessage({
-        hasError: hasFoldersError,
-        message: foldersErrorMessage,
-        folders: pageItems,
-        query,
-      });
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CopyView/CopyViewProvider.tsx:99]: Conditionally returning `undefined` here was the root cause of the memo issue, moved the call to `getListFoldersResultsMessage` itself inside `FoldersMessageControl` and removed the init flag

File: packages/react-storage/src/components/StorageBrowser/components/DropdownMenu.tsx [hunk 4]
```diff
@@ -96,6 +96,7 @@ export function DropdownMenu({
                 icon={icon}
                 label={label}
                 onClick={() => {
+                  setIsOpen(false);
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/components/DropdownMenu.tsx:99]: Addresses a regression in the ActionsList composable migration, the menu was not closing on item select in composed StorageBrowser usage

**Review by cshfang — APPROVED** [review]

**Review by jordanvn — APPROVED** [review]

**Pull request merged**